internal/db/schema/migrations/oss/postgres_96_07_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package oss_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/common"
+	"github.com/hashicorp/boundary/internal/db/schema"
+	"github.com/hashicorp/boundary/internal/db/schema/migration"
+	"github.com/hashicorp/boundary/internal/db/schema/migrations/oss/internal/hook96007"
+	"github.com/hashicorp/boundary/testing/dbtest"
+	"github.com/stretchr/testify/require"
+)
+
+// cases to look at [NA = not-applicable | P = pass | F = fail] - 'FindProblem' query should find all `F` cases
+// [F] Global Role (r_globala_96007) - Grant This, Descendants, individual org, and individual project
+//   - need to remove individual org
+//   - need to remove individual proj
+//
+// [F] Global Role (r_globalb_96007) - Grant This, Children, individual org, and individual project
+//   - need to remove individual org
+//
+// [P] Global Role (r_globalc_96007) - Grant role's self scope ID
+//   - switch to `this` automagically
+//
+// [F] Org Role (r_orgaa___96007) - Grant This, Children, and individual project
+//   - need to remove individual project
+//
+// [P] Org Role (r_orgab___96007) - Grant role self scope ID
+//   - switch to `this` automagically
+//
+// [P] Project Role (r_prjaa___96007) - Grant self scope ID
+//   - switch to `this` automagically
+//
+// [NA] Org Role - Grant Children AND individual org (different org)
+//   - this cannot be created
+//
+// [NA] Org Role - Grant Children AND individual project (from different org)
+//   - this cannot be created
+//
+// [NA] Project Role (p_pA___96007) - Grant individual project (different proj)
+//   - this cannot be created
+//
+// [NA] Global Role - Grant Descendants and Children
+//   - this cannot be created
+func TestMigrationHook_FindIllegal(t *testing.T) {
```

Comment:
Perhaps make this function name more specific to the migration hook it relates to, since `FindIllegalAssociations` is implemented in more than one package: 

```suggestion
func TestMigrationHook96007_FindIllegal(t *testing.T) {
```